src/recommend.rs

```diff
@@ -0,0 +1,76 @@
+use std::ops::Index;
+
+use reqwest::Url;
+use serde_derive::{Deserialize, Serialize};
+
+type ConnectivityResponse = Vec<Nodes>;
+
+#[derive(Default, Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct Nodes {
+    pub public_key: String,
+    pub channels: i64,
+    pub capacity: i64,
+}
+
+#[derive(Default, Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Node {
+    #[serde(rename = "public_key")]
+    pub public_key: String,
+    pub alias: String,
+    pub sockets: String,
+    #[serde(rename = "active_channel_count")]
+    pub active_channel_count: i64,
+    pub capacity: String,
+}
```

Comment:
We're leaving a lot to be desired by ignoring rust's strong typing system. None of these values are validated, which they could be by using e.g. `ln_types::P2pAddress`. Instead I see data transformation in the javascript.

Before, our view didn't do any business logic. Adding business logic to the presentation layer is a regression. Shouldn't typing and business logic live in this controller instead?

Reply:
> We're leaving a lot 


Its hard for me to discern exactly what changes we want -- some of my confusion might come from still learning rust. 
Can you give specific changes? Are you talking about the sockets?

Reply:
1. Use the type most specific for each field. see https://github.com/chaincase-app/nolooking/pull/71/files/d24e73f683a62d91f1f50a5f1bdff92c802ca7a6#r1038863893
2. only present data in the view. Do not transform it or assume its shape. Assume the UI understands no more about the data than the interface you give it. I don't see a good reason for p2paddres parsing to happen in the javascript when we can do it here with types that are already well tested.